src/python/grpcio/grpc/_channel.py

```diff
@@ -496,7 +515,11 @@ def future(self, request, timeout=None, metadata=None, credentials=None):
         call.set_credentials(credentials._credentials)
       event_handler = _event_handler(state, call, self._response_deserializer)
       with state.condition:
-        call.start_client_batch(cygrpc.Operations(operations), event_handler)
+        call_error = call.start_client_batch(cygrpc.Operations(operations),
+            event_handler)
+        if call_error != cygrpc.CallError.ok:
+          return _Rendezvous(
+              _call_error_RPCstate(call_error, metadata), None, None, deadline)
```

Comment:
Normally I'm the guy who's totally against all mutation, but... I think this code is easier to reason about if there remains exactly one `_RPCState` per RPC. I think you should change `_call_error_RPCstate` to mutate the `_RPCState` passed to it rather than create a new instance.